### DISCUSSION TITLE
Help with MastodonVersionError

### DISCUSSION THREAD

**Discussion by carolmb** [mastodon/mastodon#24765] [Q&A] [answered]
Hello. I've been using the API these days, but since yesterday the following error is occurring and I can't find any record of it. 

> 
> ---> 17 user_profile = api1.account_lookup("{}@{}".format(user_name, user_server))
> 
> File [d:\Documentos\yy-projects\mastodon-proj\.venv\Lib\site-packages\decorator.py:232](file:///D:/Documentos/yy-projects/mastodon-proj/.venv/Lib/site-packages/decorator.py:232), in decorate..fun(*args, **kw)
>     230 if not kwsyntax:
>     231     args, kw = fix(args, kw, sig)
> --> 232 return caller(func, *(extras + args), **kw)
> 
> File [d:\Documentos\yy-projects\mastodon-proj\.venv\Lib\site-packages\mastodon\utility.py:44](file:///D:/Documentos/yy-projects/mastodon-proj/.venv/Lib/site-packages/mastodon/utility.py:44), in api_version..api_min_version_decorator..wrapper(function, self, *args, **kwargs)
>      42 major, minor, patch = parse_version_string(version)
>      43 if major > self.mastodon_major:
> ---> 44     raise MastodonVersionError(f"Version check failed (Need version {version})")
>      45 elif major == self.mastodon_major and minor > self.mastodon_minor:
>      46     raise MastodonVersionError(f"Version check failed (Need version {version})")
> 
> **MastodonVersionError: Version check failed (Need version 3.4.0)**

Since I'm not familiar with where exactly I should ask about this kind of stuff if there is a better place, let me know. I can delete this one and repost where it is more suitable. 

Thank you in advance. 

More information:
I'm using python (3.11.3)  and mastodon.py (1.8.1).

**Comment by trwnh** [comment, marked as answer]
https://github.com/halcy/Mastodon.py

**Comment by tshrinivasan** [comment]
```
import mastodon
from mastodon import Mastodon
m = Mastodon( version_check_mode="none",access_token="TOKEN", api_base_url="https://URL")
m.status_post("Test post via API using OAuth", visibility="public")
```
set version_check_mode="none" so that we can disable checking for the version.

Reference - 
https://mastodonpy.readthedocs.io/en/1.4.6/index.html?highlight=version_check_mode#mastodon.Mastodon.__init__

The mastodon_version parameter can be used to specify the version of Mastodon that Mastodon.py will expect to be installed on the server. The function will throw an error if an unparseable Version is specified. If no version is specified, Mastodon.py will set mastodon_version to the detected version.

The version check mode can be set to “created” (the default behaviour), “changed” or “none”. If set to “created”, Mastodon.py will throw an error if the version of Mastodon it is connected to is too old to have an endpoint. If it is set to “changed”, it will throw an error if the endpoints behaviour has changed after the version of Mastodon that is connected has been released. If it is set to “none”, version checking is disabled.
